library/Makefile

```diff
@@ -91,9 +91,9 @@ shared: libmbedcrypto.$(DLEXT) libmbedx509.$(DLEXT) libmbedtls.$(DLEXT)
 # tls
 libmbedtls.a: $(OBJS_TLS)
 	echo "  AR    $@"
-	$(AR) -rc $@ $(OBJS_TLS)
+	$(AR) rc $@ $(OBJS_TLS)
```

Comment:
GNU ar may treat the dash as optional, but there are other implementations that require it, for example ARMcc 5 and 6. I'm surprised about LLVM ar here, its documentation states that the dash is optional, and I thought non-dash usage had gone out of fashion. Since both ARMcc and LLVM are relevant platforms for us, I fear we need a way to make the dash a parameter. Something like
```
# Set AR_DASH= (empty string) if your ar implentation does not accept it (e.g. llvm-ar)
AR_DASH = -
```
and
```
$(AR) $(AR_DASH)s $@
```

Reply:
```
AR_DASH ?= -
```
As a simple workaround?